.github/workflows/ci.yml

```diff
@@ -170,7 +170,7 @@ jobs:
     - name: Install Rust
       uses: actions-rs/toolchain@v1
       with:
-        toolchain: 1.54.0  # MSRV
+        toolchain: nightly
```

Comment:
Why was this changed?

Our clippy toolchain for PRs is set to MSRV so contributors dont't get a CI failure for something they aren't responsible for.

Reply:
> Why was this changed?
> 
> Our clippy toolchain for PRs is set to MSRV so contributors dont't get a CI failure for something they aren't responsible for.

There are two separate concerns here
- configuring clippy to not make suggestions which use syntax not supported by the MSRV
- pinning the clippy version so builds are reproducible. (Ie new 'errors' aren't introduced only because of a clippy update)

The current config seems to conflate the two.

The first case is handled by the config in the `clippy.toml` file, regardless of which toolchain is used.

But if you're targeting the second use case, you might as well pin to a very recent toolchain- no reason to pin to the MSRV

What are the intended semantics?

Reply:
The intention is
> pinning the clippy version so builds are reproducible. (Ie new 'errors' aren't introduced only because of a clippy update)

The reason I don't pin to a more recent toolchain is this makes it easier to track and update, just update it when we update MSRV.

Reply:
reverted